libopenage/util/profiler.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright 2015-2015 the openage authors. See copying.md for legal info.
+
+#include "profiler.h"
+#include "../engine.h"
+
+#include <chrono>
+#include <epoxy/gl.h>
+#include <iostream>
+
+namespace openage {
+namespace util {
+
+Profiler::~Profiler() {
+	this->unregister_all();
+}
+
+void Profiler::register_component(std::string com, color component_color) {
+	if (this->registered(com)) {
+		return;
+	}
+
+	component_time_data cdt = component_time_data();
```

Comment:
whoops, didn't check that. I think the default constructor for `component_time_data` is already run, so you can just `component_time_data cdt;`